tools/dm/data-synchronization-features.md

```diff
@@ -0,0 +1,452 @@
+---
+title: 数据同步功能
+summary: DM 提供的功能及其配置介绍
+category: tools
+---
+
+# 数据同步功能
+
+这个文档详细的介绍了 DM 提供的数据同步功能，以及相关的配置选项
+
+## Table routing
+
+table routing 提供将上游 MySQL/MariaDB 实例的某些表同步到下游指定表的功能。
+
+> **注意:**
+>
+> - 不支持对同一个表设置多个不同的路由规则。
+> - schema 的匹配规则需要单独设置，用来同步 create/drop schema xx，例如下面的 rule-2。
+
+### 参数配置
+
+```yaml
+routes:
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+### 参数解释
+
+将根据 [`schema-pattern`/`table-pattern`](/tools/dm/table-selector.md) 匹配上该规则的上游 MySQL/MariaDB 实例的表同步到下游的 `target-schema`/`target-table`
+
+### 使用示例
+
+下面展示了三个不同场景下的配置示例。
+
+#### 分库分表合并
+
+假设存在分库分表场景，需要将上游两个 MySQL 实例的表 `test_{1,2,3...}`.`t_{1,2,3...}` 同步到下游 TiDB 的一张表 `test`.`t`。
+
+为了同步到下游实例的表 `test`.`t` 需要创建两个 table routing 规则：
+
+- `rule-1` 用来同步匹配上 `schema-pattern: "test_*"` 和 `table-pattern: "t_*"` 的表的 DMLs/DDLs 到下游的 `test`.`t`。
+- `rule-2` 用来同步匹配上 `schema-pattern: "test_*"` 的库的 DDLs，例如 `create/drop schema xx`。
+
+> **注意:**
+>
+> - 如果下游 TiDB `schema: test` 已经存在， 并且不会被删除则可以省略 `rule-2`。
+> - 如果下游 TiDB `schema: test` 不存在，只设置了 `rule_1` 则同步会报错 `schema test doesn't exist`。
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+#### 分库合并
+
+假设存在分库场景，将上游两个 MySQL 实例 `test_{1,2,3...}`.`t_{1,2,3...}` 同步到下游 TiDB 的 `test`.`t_{1,2,3...}`，创建一条路由规则即可：
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+#### 错误的 table routing
+
+假设存在下面两个路由规则，`test_1_bak`.`t_1_bak` 可以匹配上 `rule-1` 和 `rule-2`， 违反 table 路由的限制而报错。
+
```

Comment:
we have two errors in the following route rules:
1. we don't provide a valid schema level route rule
2. table level route rules have conflict

should we add a schema level route rule first?